src/util/file-filter.js

```diff
@@ -0,0 +1,109 @@
+/* @flow */
+import path from 'path';
+
+import minimatch from 'minimatch';
+
+import {createLogger} from './logger';
+
+const log = createLogger(__filename);
+
+// Use this function to mimic path.resolve without resolving to absolute path.
+export const normalizeResolve = (file: string): string => {
```

Comment:
I don't fully understand why we need this function. Is it just for convenience so that someone can use an ignore path of `src/package/../data/*.json` instead of `src/data/*.json`? And the trailing slash protection is also for convenience? I feel like we shouldn't impose too many constraints on the `ignoreFiles` logic since that's up to the developer to use for customization.